bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java

```diff
@@ -194,165 +219,124 @@ private boolean getDevices() {
      * Subscribe to events and store the subscription ID needed for long polling
      *
      * Method is synchronous.
+     * 
+     * @throws ExecutionException
+     * @throws TimeoutException
+     * @throws InterruptedException
+     * 
+     * @return Subscription id
      */
-    private void subscribe(BoschHttpClient httpClient) {
-        logger.debug("Sending subscribe request to Bosch");
+    private String subscribe(BoschHttpClient httpClient)
+            throws InterruptedException, TimeoutException, ExecutionException {
+        String url = httpClient.createUrl("remote/json-rpc");
+        JsonRpcRequest request = new JsonRpcRequest("2.0", "RE/subscribe",
+                new String[] { "com/bosch/sh/remote/*", null });
+        logger.debug("Subscribe: Sending request: {} - using httpClient {}", gson.toJson(request), httpClient);
+        Request httpRequest = httpClient.createRequest(url, POST, request);
+        SubscribeResult response = httpClient.sendRequest(httpRequest, SubscribeResult.class);
+
+        logger.debug("Subscribe: Got subscription ID: {} {}", response.getResult(), response.getJsonrpc());
+        String subscriptionId = response.getResult();
+        return subscriptionId;
+    }
 
-        class SubscribeListener extends BufferingResponseListener {
-            private BoschSHCBridgeHandler bridgeHandler;
+    private void scheduleLongPoll(BoschHttpClient httpClient, String subscriptionId, long delay) {
```

Comment:
Nice pattern to scheduleLongPolls within the initial scheduler.execute call.
I think I will reuse the same pattern if a pairing is necessary to replace the while loop in ``checkAccessAndPairIfNecessary()``.

Reply:
Done with ``scheduleInitialAccess()``